esp/GBPlay/main/hardware/wifi.c

```diff
@@ -153,6 +153,25 @@ void wifi_scan(wifi_ap_info* ap_list, uint16_t* ap_count)
     xSemaphoreGive(s_wifi_lock);
 }
 
+static void _disconnect()
+{
+    xEventGroupClearBits(s_wifi_event_group, 0xFF);
+
+    if (wifi_is_connected())
+    {
+        esp_wifi_disconnect();
+
+        // Wait for disconnect
```

Comment:
Waiting for the disconnect is more stable, particularly since it is called in `wifi_connect()` before attempting to reconnect. We don't want to start reconnecting before the previous disconnect is finished.